adapters/repos/db/vector/hnsw/maintainance.go

```diff
@@ -26,7 +26,7 @@ const (
 // function growing the index of the hnsw struct. It does not do any locking on
 // its own, make sure that this function is called from a single-thread or
 // locked situation
-func (h *hnsw) growIndexToAccomodateNode(id int, logger logrus.FieldLogger) error {
+func (h *hnsw) growIndexToAccomodateNode(id int64, logger logrus.FieldLogger) error {
```

Comment:
whole file: see above, all node ids should be represented by `uint64`, removing the need for the conversions.